tests/unit/test_artifactory_path.py

```diff
@@ -910,20 +935,38 @@ def test_deploy_file(self):
             file.write("I am a test file")
 
             constructed_url = f"{path}{test_file.name};{matrix_parameters}"
-            responses.add(
-                responses.PUT, constructed_url, status=200, match_querystring=True
-            )
-
-            path.deploy_file(
-                test_file,
-                explode_archive=True,
-                explode_archive_atomic=True,
-                parameters={
-                    "deb.architecture": "amd64",
-                    "deb.component": "contrib",
-                    "deb.distribution": ["dist1", "dist2"],
-                },
-            )
+            responses.add(responses.PUT, constructed_url, status=200)
+
+            # can't use pytest.mark.parametrize with unittest classes
+            for quote_params in (None, True, False):
+                with mock.patch(
```

Comment:
Let's assert against pure values, without checking `q.call_count`

Reply:
This comment is on the `with mock.patch(` line, was this meant for line 967? or a different line?

When you say assert against pure values, what does that mean exactly? I apologize if I'm misunderstanding.

Reply:
Let's remove mock and it'll fix the conversation :) 

I think we need to add some quoted symbol as a value or as a key in `parameters` below` and assert against to different values based on `quote_params` value

Reply:
well.. this is why I wanted to assert the calls, so that we don't need to worry about the specific characters we use (since this test is not for that, it's just for ensuring that when deploy is called, it passes through the `quote_parameters` value to the underlying function correctly.

I can certainly update it as you requested, but I wonder if this test is the right place to check implementation of a different function than the one we're testing.

Let me know if you still prefer that I will change it!

Reply:
I have done as you asked and removed the wrapper around `quote`, and added test cases to `parameters`, in commit https://github.com/devopshq/artifactory/pull/409/commits/a6acebae54cde683e43d47cb37b0fe700ef489f2. 

The changes needed were significant, and I ran into several tricky issues along the way, but hopefully this is more what you were looking for? Please let me know.